autopush/router/gcm.py

```diff
@@ -43,18 +45,20 @@ def register(self, uaid, router_data, router_token=None, *kwargs):
         if "token" not in router_data:
             raise self._error("connect info missing GCM Instance 'token'",
                               status=401)
+        if not (router_token == router_data['token'] == self.senderID):
```

Comment:
It's a bit complex and a hang-over from when we were going to use multiple senderids to deliver content. That got dropped, but it's a useful check for situations such as this. "Router_Token" is a client side, hard coded. value that is the GCM SenderID/Project Number. It needs to match much in the same way as our UAID needs to match. I could pare the value out of the router data if that's a concern.